### ISSUE TITLE
[Feature]: The ability to group content pieces

### ISSUE THREAD

**Issue by jack-white-securecheck** [Bartek532/syncreads#88]
### Feature description

A suggestion to have the ability to group content pieces, e.g. I am a software engineer that can see myself using the application for all the documentation that I frequently use, but I have a few articles from NextJS, a few from Supabase and a few from AWS. Then to have the ability to group those together could be really handy. That way, my content won't become too cluttered and I can easily find articles that I previously read.

### Additional Context

Additional details here...

**Comment by Bartek532** [comment]
Do you mean introducing some kind of tags, which could be added to the document? It's in the plan, even with generating them with AI, what do you think about it?

**Comment by jack-white-securecheck** [comment]
Ah yeah that would create the same concept, where you can 'group' content pieces based on a common interest, so by doing it via tags or grouping it would result in the came outcome. Also, by using tags, I suppose you can have multiple tags on a content piece? 

Doing it with AI can be a nice addition, so then when syncing a content piece, the AI would provide you with a suggested tag? And then you can leave it or change it to your own tag?

**Comment by Bartek532** [comment]
Yes, I was thinking about the solution where user can add multiple tags and AI is used only to provided helpful examples, but the final decision will be always on the user.

**Comment by jack-white-securecheck** [comment]
Yeah that would be a good feature! Looking forward to seeing that in action!
